### PR TITLE
[VA] Summary Experience for VA/Skills

### LINKED ISSUE #2057: Summary skill based on Action integration
## User Story
As ...
I want ...
so that ...

## Acceptance Criteria 
*List your requirements.*
- [ ] ...
- [ ] ...
- [ ] ...

## Resources
*Provide any screenshots or conversation mockups to support your request.*
*Learn how to mock up bot conversations with [.chat files](http://aka.ms/bfchatfile).*

## Notes
*Provide any additional information.*

#### Is your feature request related to a problem? Please describe. 
Create the summary skill based on new action support.

### PR DESCRIPTION
**Pull request by xhr0804**
### Purpose
Close #2057 

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

Create a prototype for VA.SummarySkill,  with a brief summary of Calendar/Email/Todo skill currently.
The project is still in progress now, and more features will be added.
It currently works like:
image

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

Will make the project more flexible and robust, and add more features to improve UX.

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [ ] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

**Review by bobokids — COMMENTED** [review]
should not change.

File: lib/csharp/Directory.Build.props [hunk 1]
```diff
@@ -8,7 +8,7 @@
   </PropertyGroup>
 
   <PropertyGroup>
-    <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
+    <TreatWarningsAsErrors>False</TreatWarningsAsErrors>
```
Comment (bobokids) [inline lib/csharp/Directory.Build.props]: should not change

File: skills/src/csharp/calendarskill/calendarskill/Dialogs/MainDialog.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@
 using Microsoft.Bot.Builder.Solutions.Responses;
 using Microsoft.Bot.Connector;
 using Microsoft.Bot.Schema;
+using Microsoft.CodeAnalysis.CSharp;
```
Comment (bobokids) [inline skills/src/csharp/calendarskill/calendarskill/Dialogs/MainDialog.cs]: Need this line?

File: skills/src/csharp/calendarskill/calendarskill/Dialogs/CalendarSummaryDialog.cs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+            var getMeetings = new WaterfallStep[]
+            {
+                GetAuthToken,
+                AfterGetAuthToken,
+                GetMeetings
+            };
+
+            // Define the conversation flow using a waterfall model.
+            AddDialog(new WaterfallDialog(Actions.GetMeetings, getMeetings) { TelemetryClient = telemetryClient });
```
Comment (bobokids) [inline skills/src/csharp/calendarskill/calendarskill/Dialogs/CalendarSummaryDialog.cs]: maybe GetMeetingSummary etc.

**Comment by darrenj** [comment]
I'd love to move this to a tutorial/sample for our next release if we have time. I..e show how you can write a Dialog that can invoke the skills as we show here to retrieve data. @lzc850612 perhaps this could be a test case for the SDK port changes you are working on? i..e if we can achieve the same scenario then the SDK changes are likely giving us most of what we need?

I don't think this is something that is best suited to an entire skill per-se but a sample to start with and we can then explore further. Perhaps something we seek to add to the Enterprise Template (your day today experience. @Dewain27

**Comment by lzc850612** [comment]
> I'd love to move this to a tutorial/sample for our next release if we have time. I..e show how you can write a Dialog that can invoke the skills as we show here to retrieve data. @lzc850612 perhaps this could be a test case for the SDK port changes you are working on? i..e if we can achieve the same scenario then the SDK changes are likely giving us most of what we need?
> 
> I don't think this is something that is best suited to an entire skill per-se but a sample to start with and we can then explore further. Perhaps something we seek to add to the Enterprise Template (your day today experience. @Dewain27

yes the new SDK implementation should support this. I'll make sure that I test this scenario when integrating their change into our Solutions library.

**Comment by darrenj** [comment]
Would be good to sync with @lzc850612 as we update to the latest Skill packages to make the appropriate changes and then I would suggest we create a documentation item on how to build a summary experience invoking multiple skills which shows the key code blocks and perhaps references a sample in the [Sample folder](https://github.com/microsoft/botframework-solutions/tree/master/samples/csharp) rather than making this a separate Skill. @ryanlengel  would the samples folder be the right location.

We can then close this PR and create a new one to create the docs and sample folder.

**Pull request closed without merging**